subprojects/docs/src/docs/userguide/building_cpp_projects.adoc

```diff
@@ -0,0 +1,275 @@
+It borrows several conventions from Apache Maven.
+In particular, it uses the same default directory pattern for source files and headers.
+
+We will look at C\++ projects in detail in this chapter, but most of the topics apply to other supported native languages as well.
+If you don’t have much experience with building native projects with Gradle, take a look at the C++ tutorials for step-by-step instructions on how to build various types of basic C++ projects as well as some common use case.
+
+The C++ plugins covered in this chapter were https://blog.gradle.org/introducing-the-new-cpp-plugins[introduced in 2018] and we recommend users to use those plugins over <<native_software.adoc#native_binaries,the older Native plugins>>.
+
+== Introduction
+
+The simplest build script for a C\++ project applies the C++ application plugin or the C++ library plugin and optionally sets the project version:
+
+
+.Applying the C++ Plugin
+====
+include::sample[dir="userguide/cpp/basic/groovy",files="build.gradle[tags=apply-cpp-plugin]"]
+include::sample[dir="userguide/cpp/basic/kotlin",files="build.gradle.kts[tags=apply-cpp-plugin]"]
+====
+
+By applying any of C++ plugin, you get a whole host of features:
+
+ * A `compileDebugCpp` and `compileReleaseCpp` tasks that compiles the C++ source files under _src/main/cpp_ for the well-known debug and release build type respectively.
+ * A `linkDebug` and `linkRelease` tasks that links the compiled C++ object files into an executable for applications or shared library for libraries with shared linkage for the well-known debug and release build type respectively.
+ * A `createDebug` and `createRelease` tasks that assemble the compiled C++ object files into a static library for libraries with shared linkage for the well-known debug and release build type respectively
+
+This isn’t sufficient to build any non-trivial C++ project - at the very least, you’ll probably have some file dependencies.
+But it means that your build script only needs the information that is specific to _your_ project.
+
+// TODO: Add base plugin link
+The C++ plugins also integrate the above tasks into the standard Lifecycle Base Plugin lifecycle tasks:
+
+ * `linkDebug` is attached to `assemble`
+
+The rest of the chapter explains the different avenues for customizing the build to your requirements. You will also see later how to adjust the build specifically for libraries and applications.
+
+== Build variants
+Gradle is able to build a number of different native binary variants.
+They can be differentiated on build type footnote:[For example, debug or release], platform footnote:[For example, Windows x86 or macOS x86-64] and linkage footnote:[For example, shared or static linkage] (library-only).
+For each of these 'variant dimensions', tasks, binaries and configuration are created.
```

Comment:
Is there a default variant? I'm thinking of the way that the "main" source set in Java builds results in a `classes` task rather than `mainClasses`, whereas the "test" source set results in a `testClasses` task.

Reply:
We should be following the same naming. For example, we have `compileDebugCpp` and `compileReleaseCpp` for the main component and `compileTestCpp` for the test component.

Reply:
So is `test` a build type?

Reply:
Yes and no. The `main` part for the primary component is absent. Any other component will have its name in there. The test component only has `debug` build type. Single value dimensions are also excluded from the name. I will see if I can add a bit more context to the introduction section.